src/App.tsx

```diff
@@ -1,24 +1,299 @@
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<number[]>([]);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isLoadingTodo, setIsLoadingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    async function fetchTodos() {
+      try {
+        const data = await getTodos(USER_ID);
+
+        setTodos(data);
+        setUptatedTodos(data);
+        setTodosNotCompleted(
+          data.filter(todo => todo.completed === false).length,
+        );
+      } catch (error) {
+        setErrorMessage('Unable to load a todo');
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    fetchTodos();
+    setIsTodoDeleted(false);
+  }, [isTodoDeleted, tempTodo]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  useEffect(() => {
+    const filteredTodos = todos.filter(todo => {
+      switch (sortBy) {
+        case 'completed':
+          return todo.completed === true;
+        case 'active':
+          return todo.completed === false;
+        case 'all':
+          return todo;
+        default:
+          return todo;
+      }
+    });
+
+    setUptatedTodos(filteredTodos);
+  }, [sortBy]);
+
+  const deleteOneTodo = async (todoId: number) => {
+    try {
+      await deleteTodo(USER_ID, todoId);
+    } catch (error) {
+      setErrorMessage('Unable to delete a todo');
+    }
+
+    setIsTodoDeleted(true);
+    setSelectedTodo(prevSelectedTodo => [...prevSelectedTodo, todoId]);
+  };
+
+  const handleKeyDown = async (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (!inputValue) {
+        setErrorMessage("Title can't be empty");
+      }
+
+      setIsLoadingTodo(true);
+
+      setTempTodo({
+        id: 0,
+        userId: 11361,
+        title: inputValue,
+        completed: false,
+      });
+
+      const todo = {
+        id: 0,
+        userId: 11361,
+        title: inputValue,
+        completed: false,
+      };
+
+      setInputValue('');
+
+      try {
+        await addTodo(USER_ID, todo);
+      } catch (error) {
+        setErrorMessage('Unable to add a todo');
+      } finally {
+        setIsLoadingTodo(false);
+        setTempTodo(null);
+      }
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
```

Comment:
Overall your component here is too long, your components should be max 200 lines long (the shorter the better), split some of the jsx return here into separate components